llarp/util/thread/threadpool.h

```diff
@@ -5,48 +5,51 @@
 #include <util/thread/queue.hpp>
 #include <util/thread/thread_pool.hpp>
 #include <util/thread/threading.hpp>
+#include <util/types.hpp>
 
 #include <absl/base/thread_annotations.h>
 #include <memory>
 #include <queue>
 
+struct llarp_threadpool;
+
+#ifdef __cplusplus
 struct llarp_threadpool
 {
   std::unique_ptr< llarp::thread::ThreadPool > impl;
-  std::unique_ptr< llarp::thread::Queue< std::function< void(void) > > > jobs;
-  const pid_t callingPID;
-
-  llarp_threadpool(int workers, llarp::string_view name)
-      : impl(std::make_unique< llarp::thread::ThreadPool >(workers,
-                                                           workers * 128, name))
-      , jobs(nullptr)
-      , callingPID(0)
-  {
-  }
 
-  llarp_threadpool()
-      : jobs(new llarp::thread::Queue< std::function< void() > >(128))
-      , callingPID(llarp::util::GetPid())
+  llarp_threadpool(int workers, llarp::string_view name,
+                   size_t queueLength = size_t{1024})
+      : impl(std::make_unique< llarp::thread::ThreadPool >(
+          workers, std::max(queueLength, size_t{32}), name))
   {
-    jobs->enable();
   }
 
   size_t
-  size() const
+  size() const;
+
+  size_t
+  pendingJobs() const;
+
+  size_t
+  numThreads() const;
+
+  /// try to guess how big our job latency is on this threadpool
+  llarp_time_t
+  GuessJobLatency(llarp_time_t granulairty = 1000) const;
```

Comment:
A high level description of granularity would be useful. Also, typo (granulairty)